controllers/datamovement_controller_test.go

```diff
@@ -60,23 +110,54 @@ var _ = Describe("Data Movement Test" /*Ordered, (Ginkgo v2)*/, func() {
 				},
 				UserId:  uint32(os.Geteuid()),
 				GroupId: uint32(os.Getegid()),
-				Cancel:  false,
+				Cancel:  dmCancel,
 			},
 		}
-	})
 
-	JustBeforeEach(func() {
+		cm = &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      configMapName,
+				Namespace: configMapNamespace,
+				Labels: map[string]string{
+					testLabelKey: testLabel,
+				},
+			},
+			Data: cmData,
+		}
+
+		// Create CM and verify label
+		if createCm {
+			Expect(k8sClient.Create(context.TODO(), cm)).To(Succeed())
+			Eventually(func(g Gomega) string {
+				g.Expect(k8sClient.Get(context.TODO(), client.ObjectKeyFromObject(cm), cm)).To(Succeed())
+				return cm.Labels[testLabelKey]
+			}).Should(Equal(testLabel))
+		}
+
+		// Create DM and verify label
 		Expect(k8sClient.Create(context.TODO(), dm)).To(Succeed())
+		Eventually(func(g Gomega) string {
+			g.Expect(k8sClient.Get(context.TODO(), client.ObjectKeyFromObject(dm), dm)).To(Succeed())
+			return dm.Labels[testLabelKey]
+		}).Should(Equal(testLabel))
+
 	})
 
-	JustAfterEach(func() {
-		Expect(k8sClient.Delete(context.TODO(), dm)).To(Succeed())
+	AfterEach(func() {
+		// Remove datamovement
+		k8sClient.Delete(context.TODO(), dm)
```

Comment:
Expect To Succeed?

Reply:
We might not always have a dm to delete (see [here](https://github.com/NearNodeFlash/nnf-dm/blob/dm-progress/controllers/datamovement_controller_test.go#L410)). We really only care that the Get() comes up empty. So testing for the end result is more flexible.

Reply:
I was following the aspect from a declarative API, I don't care about the how, I just care about the end result. I see your point though. I'll add checks for deleting dm/cm.